app/src/main/java/ch/onepass/onepass/utils/TagNormalizer.kt

```diff
@@ -0,0 +1,41 @@
+package ch.onepass.onepass.utils
+
+import ch.onepass.onepass.model.event.EventTag
+
+/**
+ * Utility object for normalizing and comparing tags with case-insensitive matching.
+ *
+ * Provides functionality to match event tags against target tags, handling both enum names and
+ * display values with normalization.
+ */
+object TagNormalizer {
+  /**
+   * Checks if an event tag matches any of the target tags, considering both enum names and display
+   * values.
+   *
+   * @param eventTag The tag from an event to check for matches
+   * @param targetTags The set of tags to match against (typically from filter selections)
+   * @return true if the event tag matches any of the target tags, false otherwise
+   */
+  fun matchesTag(eventTag: String, targetTags: Set<String>): Boolean {
+    val normalizedEventTag = eventTag.trim()
+
```

Comment:
Missing edge case handling, what if `eventTag` is a whitespace or empty string ? After `.trim()`, `normalizedEventTag` would be empty. 

```suggestion
    if (normalizedEventTag.isEmpty()) return false
```

Reply:
Agree with you 100%.
Fixed!